astropy/units/quantity.py

```diff
@@ -1271,15 +1271,15 @@ def _repr_latex_(self):
         # with array2string
         pops = np.get_printoptions()
         try:
-            formatter = {'float_kind': Latex.format_exponential_notation}
+            formatter = {'float_kind': Latex.format_exponential_notation_with_np_precision}
```

Comment:
Instead of defining a new function in `Latex`, I would just define it here (another problem solved by adding a layer of indirection...)
```
format_spec = '.{}g'.format(pops['precision'])
def float_formatter(value):
    return Latex.format_exponential(value, format_spec=format_spec)

try:
    formatter = {'float_kind': float_formatter}
    ...
```
(and of course replace for `style=...` as well)
    return